File: backend/models/file.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID

class FileSendBase(BaseModel):
    receiver_id: UUID
    encrypted_payload: str
    kyber_ciphertext: str
    dilithium_signature: str

class FileSendCreate(FileSendBase):
    pass

class FileResponse(FileSendBase):
    id: UUID
    sender_id: UUID
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

File: flutter_app/lib/models/file.dart
class SecureFile {
  final String id;
  final String senderId;
  final String receiverId;
  final String status;
  final DateTime createdAt;

  SecureFile({
    required this.id,
    required this.senderId,
    required this.receiverId,
    required this.status,
    required this.createdAt,
  });

  factory SecureFile.fromJson(Map<String, dynamic> json) {
    return SecureFile(
      id: json['id'],
      senderId: json['sender_id'] ?? '',
      receiverId: json['receiver_id'] ?? '',
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}
